src/west/app/project.py

```diff
@@ -924,6 +924,30 @@ def fetch_strategy(self, args):
         else:
             return 'smart'
 
+    def update_submodules(self, project):
+        # Updates given project submodules by using
+        # 'git submodule update --init --checkout --recursive' command
+        # from the project.path location.
+        submodules = project.submodules
+        call_location = os.path.join(self.topdir, project.path)
+
+        if submodules is None:
+            return
+
+        submodules_update_strategy = ('--rebase' if self.args.rebase
+                                      else '--checkout')
+        # For the list type, update given list of submodules.
+        if isinstance(submodules, list):
+            for submodule in submodules:
+                if isinstance(submodule, Submodule):
```

Comment:
Is there any valid situation where `isinstance(submodules, list)` is True but `sinstance(submodule, Submodule)` is False for some submodule? If yes this deserves a short comment mentioning that situation. If not then this code should crash, not silently ignore a bug. The easiest is to just remove this test and let `submodule.path` crash below. If not enough for some reason then manually throw an exception in a new `else` clause.

(Sorry I didn't spot this earlier)

Reply:
Ok, that's right, done. I think that this check was even not needed, as submodules objects integrity is previously checked, while getting submodules list from the manifest in `_get_submodules` method.